### PR TITLE
[Doc] Update aibrix documentation

### PR DESCRIPTION
**Pull request by Jeffwan**
## Pull Request Description
- Update to use Qwen open source model instead of llama
- Add development doc page
- Update rc release steps
- Polish the installation page

## Related Issues
Resolves: part of https://github.com/aibrix/aibrix/issues/390

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

### CODE REVIEW

File: docs/source/getting_started/installation.rst [hunk 1]
```diff
@@ -22,7 +22,10 @@ Stable Version
 
 .. code:: bash
 
+    # Install component dependencies
     kubectl apply -f https://github.com/aibrix/aibrix/releases/download/v0.1.1/aibrix-dependency-v0.2.0-rc.1.yaml
```
Comment (nwangfw) [inline docs/source/getting_started/installation.rst]: Shouldn't it be kubectl apply -f https://github.com/aibrix/aibrix/releases/download/v0.2.0-rc.1/aibrix-dependency-v0.2.0-rc.1.yaml? Same for the launch command below.

File: docs/source/getting_started/quickstart.rst [hunk 2]
```diff
@@ -174,10 +174,9 @@ Invoke the model endpoint using gateway api
 
     curl -v http://localhost:8888/v1/completions \
         -H "Content-Type: application/json" \
-        -H "user: test-user" \
         -H "model: meta-llama/Llama-2-7b-hf" \
```
Comment (nwangfw) [inline docs/source/getting_started/quickstart.rst]: -H "model: Qwen/Qwen2.5-7B-Instruct"  ?

File: docs/source/getting_started/installation.rst (hunk 1 above)
Comment (Jeffwan) [reply docs/source/getting_started/installation.rst]: good catch. I think that's something incorrect in last commit. Let me fix it

File: docs/source/getting_started/quickstart.rst (hunk 2 above)
Comment (Jeffwan) [reply docs/source/getting_started/quickstart.rst]: good catch. BTW, we have to use `qwen25-7b-Instruct`, aligned with value of `--served-model-name`

**Pull request merged**